### ISSUE TITLE
HTML printing tall documents fails

### ISSUE THREAD

**Issue by tresf** [qzind/tray#156]
**Version:**  2.1 branch (latest)
**HTML Content:**  [html_sample.html](https://gist.github.com/tresf/bd19db2fac7e75e74a972402a72f7f7d)
**Printer:** [PDFCreator](http://www.pdfforge.org/pdfcreator)

Settings: (`flavor: "file", data: "assets/html_sample.html"`)
 * sample.html default -- renders table [squeezed in upper left corner](https://cloud.githubusercontent.com/assets/6345473/22190186/81adf4fa-e0ef-11e6-9c84-770a84c321bc.png).
 * `size: { width: 8.5, height: 11 }` -- renders table [squeezed in upper left corner](https://cloud.githubusercontent.com/assets/6345473/22190186/81adf4fa-e0ef-11e6-9c84-770a84c321bc.png).
 * `pageWidth: 8.5` -- renders blank HTML page

![image](https://cloud.githubusercontent.com/assets/6345473/22190414/0c01b62c-e0f1-11e6-98be-886788531921.png)

**Comment by tresf** [comment]
This was due to `{ scaleConent: true }`.  Disabling it is better, but still has a bug where the right side is chopped off.

![image](https://cloud.githubusercontent.com/assets/6345473/22211791/42be3f58-e15c-11e6-8566-0fc1d9ecffed.png)

**Comment by bberenz** [comment]
> but still has a bug where the right side is chopped off.

Problem with web page width not being adjusted when changing scale for web dpi.
6571003366a741cf9d9574c1530e0989234f997f

**Comment by tresf** [comment]
@bberenz perfect.  That fixes it.

So what is left:

1.  Add intelligent way to disable `scaleConent: true` when Print HTML is clicked.  A soft warning at the top of the page is sufficient.  I believe it should only happen on first page load (e.g. store a hidden property so one can still explicitly use the checkbox if so desired).
2. It appears the scaling is still off when calculating page height (see "Actual" picture above).


![image](https://cloud.githubusercontent.com/assets/6345473/22213380/030136d0-e162-11e6-817c-499662412b10.png)

**Comment by bberenz** [comment]
For number 2, it looks like the scale to page was not accounting for the adjusted web scale (96 vs 72), should be corrected under 4e9275e0ffaf203aed868d707094caec86ee2e69

As for number 1, I'm not entirely sure what you want done.

**Comment by tresf** [comment]
> As for number 1, I'm not entirely sure what you want done.

Saner defaults without breaking the sample page... Perhaps something like:

```js
$("#pxlScale").change(function(){
   $("#pxlScale").prop('dirty', true);
});

if ($("#pxlScale").prop('checked') && !$("#pxlScale").prop('dirty')) {
   $("#pxlScale").prop('checked', false);
   displayMessage("Scale content was automatically disabled for better HTML printing.", 'alert-warning');
}


```

**Comment by bberenz** [comment]
Dirty values should be available with 60d33ab7bdf65b5b88dc84c33e01e11cb214e775
Should be easily extensible should we need this on other check boxes later

**Comment by tresf** [comment]
@bberenz this works great, thanks.

Closed via 6571003 4e9275e 60d33ab.
